src/components/cart/components/Cart.js

```diff
@@ -1,158 +1,47 @@
 import React, { Component } from 'react';
 import { Table, Badge, Button, Popconfirm, Icon } from 'antd';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 
-const data = [
-    {
-        id: '1',
-        image: '/rice.jpg',
-        price: 32,
-        product: 'Yellow Corn',
-        quantity: 2,
-    },
-    {
-        id: '2',
-        image: '/rice.jpg',
-        price: 400,
-        product: 'Beans',
-        quantity: 5,
-    },
-    {
-        id: '3',
-        image: '/rice.jpg',
-        price: 32,
-        product: 'Banana',
-        quantity: 1,
-    },
-];
+import * as actions from '../actions';
+import { getCart } from '../selectors';
 
 class Cart extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            cart: localStorage.getItem('cart') ? JSON.parse(localStorage.getItem('cart')) : [],
-            cartCount: 0,
-            totalAmount: sessionStorage.getItem('cartTotal')
-                ? sessionStorage.getItem('cartTotal')
-                : 0,
-        };
-    }
-
     componentDidMount() {
-        let total = 0;
-        const newCart = [...data];
-        newCart.some(entry => {
-            total += entry.price * entry.quantity;
-        });
-        window.localStorage.setItem('cart', JSON.stringify(newCart));
-        window.sessionStorage.setItem('cartTotal', total);
-        const count = localStorage.getItem('cart')
-            ? JSON.parse(localStorage.getItem('cart')).length
-            : 0;
-        this.setState({
-            cart: data,
-            cartCount: count,
-            totalAmount: total,
-        });
-    }
-
-    handleRemoveRow(row) {
-        let total = 0;
-        const newCart = [...this.state.cart];
-        const index = newCart.findIndex(item => row.id === item.id);
-        newCart.splice(index, 1);
-        newCart.some(entry => {
-            total += entry.price * entry.quantity;
-        });
-        window.localStorage.setItem('cart', JSON.stringify(newCart));
-        window.sessionStorage.setItem('cartTotal', total);
-        this.setState({
-            cart: newCart,
-            cartCount: newCart.length,
-            totalAmount: total,
-        });
-    }
-
-    handleQuantityChange(row, action) {
-        let total = 0;
-        const newCart = [...this.state.cart];
-        const index = newCart.findIndex(item => row.id === item.id);
-        if (action === 'add') {
-            newCart[index] = {
-                ...newCart[index],
-                quantity: newCart[index].quantity + 1,
-            };
```

Comment:
where are you handling this function

Reply:
Still working on it. I have edited status appropriately